viper.go

```diff
@@ -847,6 +849,9 @@ func (v *Viper) Sub(key string) *Viper {
 	}
 
 	if reflect.TypeOf(data).Kind() == reflect.Map {
+		subv.parents = append(v.parents, strings.ToLower(key))
+		subv.automaticEnvApplied = v.automaticEnvApplied
```

Comment:
envPrefix must also be preserved

Reply:
otherwise `1181:                 if val, ok := v.getEnv(v.mergeWithEnvPrefix(envKey)); ok {` breaks

Reply:
fixed in fd570e1 :100: